cpg_workflows/stages/stripy.py

```diff
@@ -21,6 +21,35 @@
 )
 
 
+def _stripy_config_retrieve(key: str, dataset: str | None = None) -> Any:
+    """
+    Retrieve the STRipy config for a given key. If a dataset is provided, it will first
+    check for a dataset-specific config before falling back to the global config.
+
+    **NOT to be confused with the stripy config.json file used by the stripy job**
+    """
+    if dataset and config_retrieve(['stripy', dataset, key], default=None) is not None:
+        return config_retrieve(['stripy', dataset, key], default=None)
+    else:
+        return config_retrieve(['stripy', key], default=None)
```

Comment:
```suggestion
    if dataset and (result := config_retrieve(['stripy', dataset, key], default=None)) is not None:
        return result
    return config_retrieve(['stripy', key], default=None)
```

Just a couple of nits here - 
- there's a silly linting rule somewhere about not needing an `else` after a `return`
- instead of executing `config_retrieve(['stripy', dataset, key], default=None)` twice if successful you can bank the result using a walrus operator

Reply:
Thanks Matt, good suggestion. I agree we're increasing complexity but also providing a lot more flexibility, so each dataset can freely specify their own loci.